src/design-system/theme/paletteDark.ts

```diff
@@ -138,6 +143,36 @@ const text: Partial<TypeText> = {
 };
 
 const divider = grey["20"];
+
+type ExtendedPaletteColorOptions = PaletteColorOptions & {
+  [key: string]: string | any;
+};
+
+declare module "@mui/material/styles/createPalette" {
+  export interface PaletteOptions {
+    custom?: ExtendedPaletteColorOptions;
```

Comment:
This way it should be able to autocomplete and check for typos in `custom` palette name fields.

```suggestion
    custom?: typeof custom;
```